libexec/rtld-elf/rtld_c18n.c

```diff
@@ -268,7 +268,7 @@ init_compart_stack(void **stk, compart_id_t cid)
 	 * to the top of the stack either when the compartment was last entered
 	 * or when it was last exited from, which ever occured later.
 	 */
-	stk[-1] = cheri_clearperm(stk - 1, CHERI_PERM_SW_VMEM);
```

Comment:
This is gratuitous (and less idiomatic for aggressive subobject bounds)

Reply:
You mean `stk - 1` is less idiomatic than `&stk[-1]` or the other way around?

Reply:
Yes; with sufficiently aggressive (read: ISO C violating) subobject bounds, `&a[x]` has bounds of the single element, whereas `a + x` leaves the bounds unaltered.